src/components/ChallengeModal/ChallengeModal.tsx

```diff
@@ -662,8 +628,22 @@ export class ChallengeModal extends Modal<Events, ChallengeModalProperties, any>
     update_board_size = (ev) => {
         this.syncBoardSize((ev.target as HTMLSelectElement).value);
     };
-    update_board_width = (ev) => this.upstate("challenge.game.width", ev);
-    update_board_height = (ev) => this.upstate("challenge.game.height", ev);
+
+    update_board_width = (ev) => {
+        this.setState({
+            challenge: Object.assign({}, this.state.challenge, {
+                board_width: parseInt(ev.target.value),
```

Comment:
Nice! (Also, I had no idea that `upstate` and `setState` were basically the same thing - I finally looked up the definition 😅)

Reply:
I've been nervous about that.  My understanding is that upstate handles various types of `ev` seamlessly.

I stuck with using it as long as I could, not knowing if that was all, but that pattern is copied from pre-existing below, so "ought to be OK".

Reply:
Might be related to https://github.com/online-go/online-go.com/issues/1697